### PR TITLE
Improve typing in mock.ts

### PR DESCRIPTION
**Pull request by tomekzaw**
## Summary

This PR is a small cleanup of `mock.ts` in terms of typing.

Fixes https://github.com/software-mansion/react-native-reanimated/issues/4850

## Test plan

### CODE REVIEW

**Comment by tjzel** [comment]
@tomekzaw I added all of our public API to the mocks, however I'm not sure how do we want to handle all of those functions so they are temporarily marked with a question mark with regard to their implementation.

File: src/reanimated2/mock.ts [hunk 1]
```diff
@@ -100,197 +241,175 @@ class BaseAnimationMock {
...
-    EXTEND: 'extend',
-    CLAMP: 'clamp',
-    IDENTITY: 'identity',
-  },
+const layoutReanimation = {
+  BaseAnimationBuilder: new BaseAnimationMock(),
+  ComplexAnimationBuilder: new BaseAnimationMock(),
+  Keyframe: new BaseAnimationMock(),
+  // Flip
+  FlipInXUp: new BaseAnimationMock(),
```
Comment (Latropos) [inline src/reanimated2/mock.ts:267]: WDYT about sth like this:
```tsx
Object.fromEntries(['FlipInXUp', ...].map(animationName => 
    [animationName, new BaseAnimationMock()])
```

File: .eslintrc.js [hunk 2]
```diff
@@ -33,6 +33,7 @@ module.exports = {
   },
   rules: {
     curly: 'error',
+    '@typescript-eslint/no-explicit-any': 'warn',
```
Comment (Latropos) [inline .eslintrc.js:36]: This is already on main, please merge

File: src/mock.ts [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+  isReanimated3: () => true,
+  // isConfigured: ADD ME IF NEEDED
+  enableLayoutAnimations: NOOP,
+  // getViewProp: ADD ME IF NEEDED
+};
+
+const layoutReanimation = {
+  BaseAnimationBuilder: new BaseAnimationMock(),
+  ComplexAnimationBuilder: new BaseAnimationMock(),
+  Keyframe: new BaseAnimationMock(),
```
Comment (Latropos) [inline src/mock.ts:275]: Why is this preferred over our previous solution
```tsx
].forEach((k) =>
  Object.assign(ReanimatedV2, {
    [k]: new BaseAnimationMock(),
  })
);
```
?
Anyway, you can specify type, to make sure nobody makes a typo when assigning  baseAnimations:

```suggestion
const layoutReanimation: Record<string, BaseAnimationMock> = {
  BaseAnimationBuilder: new BaseAnimationMock(),
  ComplexAnimationBuilder: new BaseAnimationMock(),
  Keyframe: new BaseAnimationMock(),
```

File: mock.js [hunk 4]
```diff
@@ -1,56 +1,10 @@
...
-
-  interpolate: NOOP,
-  interpolateColor: NOOP,
-  clamp: NOOP,
-  createAnimatedComponent: (Component) => Component,
-  addWhitelistedUIProps: NOOP,
-  addWhitelistedNativeProps: NOOP,
-};
+const Reanimated = require('./src/mock');
+const Animated = Reanimated.default;
```
Comment (tomekzaw) [inline mock.js:2]: Not sure if we need to unpack this stuff, how about just re-exporting `./src/mock`?

**Review by tomekzaw — COMMENTED** [review]
Let's merge this PR in the current shape and iterate on it.

**Review by tjzel — APPROVED** [review]

**Pull request merged**